UnitsNet.OperatorOverloads/OverloadGenerator.cs

```diff
@@ -26,10 +26,42 @@ public IEnumerable<Quantity> GetDivisionOverloads(Quantity quantity)
                     Quantity matchingQuantity = _quantities.SingleOrDefault(x => x.SiArray.EqualContent(newSiArray));
                     if (matchingQuantity != null)
                     {
-                        yield return matchingQuantity;
+                        yield return new Overload(matchingQuantity, q, quantity);
                     }
                 }
             }
         }
+
+        public IEnumerable<Overload> GetMultiplicationOverloads(Quantity quantity)
+        {
+            foreach (Quantity q in _quantities)
+            {
+                if (!quantity.Name.Equals(q.Name, StringComparison.OrdinalIgnoreCase))
+                {
+                    var newSiArray = quantity.SiArray.ElementwiseAdd(q.SiArray);
+
+                    Quantity matchingQuantity = _quantities.SingleOrDefault(x => x.SiArray.EqualContent(newSiArray));
+                    if (matchingQuantity != null)
+                    {
+                        yield return new Overload(matchingQuantity, q, quantity);
+                    }
+                }
+            }
+        }
+
+        public IEnumerable<Overload> GetOverloads(Quantity quantity)
+        {
+            var multiplications = GetMultiplicationOverloads(quantity);
+            foreach (var multiplication in multiplications)
+            {
+                yield return multiplication;
+                yield return new Overload(multiplication.Result, multiplication.Right, multiplication.Left);
```

Comment:
Is this an intended duplicate?

Reply:
It was intended, but is wrong. My thought was that multiplication is communicative, but the other multiplication will end up in the other type.